final.py

```diff
@@ -1,18 +1,217 @@
 from flask import Flask, jsonify, request
 from pymodm import connect, MongoModel, fields
 from datetime import datetime
+import base64
+import io
+import matplotlib.image as mpimg
+from matplotlib import pyplot as plt
 
 connect("mongodb://sputney13:sputney13@ds161901.mlab.com:61901/bme590final")
 app = Flask(__name__)
-app.run(host="127.0.0.1")
 
 
 class Image(MongoModel):
-    user = fields.EmailField(primary_key=True)
+    user_email = fields.EmailField(primary_key=True)
     uploaded_images = fields.ListField(field=fields.CharField())
+    image_formats = fields.ListField(field=fields.CharField())
     upload_times = fields.ListField(field=fields.DateTimeField())
     image_size = fields.ListField()
     processed_images = fields.ListField(field=fields.CharField())
     process_types = fields.ListField(field=fields.CharField())
     process_times = fields.ListField(field=fields.DateTimeField())
     user_metrics = fields.DictField()
+
+
+def validate_create_user(r):
+    """ Validates user input for posts to /image/user
+
+    Args:
+        r: dictionary containing user_email key for posting to database
+
+    Returns:
+        AttributeError: when post does not contain proper key
+        TypeError: when user_email key is not an email (does not have an @)
+
+    """
+    if "user_email" in r:
+        if "@" not in r["user_email"]:
+            raise TypeError("user_email must be an email string.")
+    else:
+        raise AttributeError("Post must be dict with user_email key.")
+
+
+@app.route("/image/user", methods=["POST"])
+def create_user():
+    """ POSTS a new image processor user (identified by email) to database
+
+    Returns:
+        200 status after posting has occurred
+
+    """
+    r = request.get_json()
+    validate_create_user(r)
+    entry = Image(r["user_email"])
```

Comment:
Do you know why "Image" is being performed on the user email? @sputney13

Reply:
https://github.com/mlp6/Medical-Software-Design/blob/master/Lectures/databases/examples/mongo/mongo-example.ipynb

From the database lecture: that's how you save a new entry to the Image class via MongoModel. Because user_email is the primary key, I don't have to specify where I'm saving to within that class- when I save that entry, it will automatically create a new entry in the database under that user's email.

Reply:
Nevermind! Just noticed it's the name of the class!